src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -37,24 +38,32 @@
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    //ServletUtils.enforceUserLogin(request, response);
+
+    ImmutableList<ImmutableMap<String, String>> groupsList = 
+        ServletUtils.getGroupsList(request.getUserPrincipal().getName());
```

Comment:
I actually changed this method to take in an Entity instead. you can add a line above this to retrieve with 
`Entity user = ServletUtils.getUserEntity(request.getUserPrincipal().getName())` then you can pass user into the getGroupsList method